### PR TITLE
Fix incorrect publish time handling

### LINKED ISSUE #7780: Publish time displaying incorrectly when site and user timezones don't match
## Bug Description
Currently, the publish time is not converted properly to/from the local time when displaying/saving the time in the calendar.

## Steps to Reproduce
First make the site and user timezone not match, for that go to the Settings menu and change the Timezone value to something that is not your current timezone.
1. Change the publish time in the calendar.
2. Save and refresh the page
3. See the time has changed

## Screenshots
time

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
## Summary
- Ensures that the date is kept in the same timezone when loading and saving the time.
- Displays timezone abbreviation in the calendar, if available.

## User-facing changes
Timezone in the calendar. The tooltip doesn't have the best visibility, perhaps we should add some shadow for it in the design system.
Screenshot 2021-06-02 at 12 50 06

## Testing Instructions
1. Set your timezone in the General Settings of WordPress to something else than UTC 0
2. Go to the editor and open the Document panel publish time calendar.
3. Verify that in the right top corner, the timezone is displayed, either UTC or your timezone.
4. Set the Publish time to something else than it was before.
5. Save and refresh the page
6. Verify the time displayed is the same as you saved previously.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. See the instructions above.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7780

### CODE REVIEW

File: assets/src/edit-story/components/form/dateTime/timeZone.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+  line-height: 30px;
+`;
+
+function TimeZone({ date }) {
+  const { timezone } = getSettings();
+
+  const zoneAbbr = timezone?.length
+    ? format(date, 'T')
+    : `UTC${getTimeZoneString()}`;
```
Comment (miina) [inline assets/src/edit-story/components/form/dateTime/timeZone.js]: @swissspidy Should this be translatable, too, 'UTC'?

File: assets/src/edit-story/components/form/dateTime/timeZone.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { getSettings, format } from '@web-stories-wp/date';
+import getTimeZoneString from '@web-stories-wp/date/src/getTimeZoneString';
```
Comment (swissspidy) [inline assets/src/edit-story/components/form/dateTime/timeZone.js]: This is not a default export, but it's available via the `getOptions()` function, which is exported.

Let's either properly export `getTimeZoneString` in `packages/date/src/index.js`, or import `getOptions` here instead.

File: assets/src/edit-story/components/form/dateTime/timeZone.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/components/form/dateTime/timeZone.js]: I don't think so. Gutenberg [doesn't do that either](https://github.com/WordPress/gutenberg/blob/fa457877f45abb692369addab76d579d5d31539d/packages/components/src/date-time/timezone.js#L18-L36).

On that note, Gutenberg's date picker has some nice logic in it (see link above):

1. Don't display timezone information if user is in the same timezone
2. Use "Coordinated Universal Time" in the tooltip if the timezone is set to UTC

Those seem like two useful enhancements to me, that I think we can apply here as well.

**Review by merapi — APPROVED** [review]
Maybe `nowrap` on that element and we are good 👍🏻 

image

BTW: I noticed that when you change the time and press enter (or click away) it will be not persisted, you need to change something more after that.
image

**Comment by csossi** [comment]
Verified in QA

**Pull request merged**